lib/post_destroyer.rb

```diff
@@ -213,6 +213,18 @@ def user_recovered
 
   private
 
+  def user_can_review_topic?
+    Guardian.new(@user).can_review_topic?(@post.topic)
+  end
+
+  def post_is_reviewable?
+    Reviewable.exists?(target: @post)
+  end
+
+  def user_deleting_own_post?
```

Comment:
I think we should avoid making this change for now since it is more of a refactor and does not affect the bug that we're trying to solve for here.